stdlib/tempfile.pyi

```diff
@@ -318,20 +324,26 @@ class SpooledTemporaryFile(IO[AnyStr]):
     def fileno(self) -> int: ...
     def flush(self) -> None: ...
     def isatty(self) -> bool: ...
+    if sys.version_info >= (3, 11):
+        def read1(self, __size: int = ...) -> AnyStr: ...
+        def readinto(self, b: WriteableBuffer) -> int: ...
+        def readinto1(self, b: WriteableBuffer) -> int: ...
```

Comment:
I'm confused by these methods.

According to the runtime, these call `self._file.read1()`, `self._file.readinto()`, `self._file.readinto1()`: https://github.com/python/cpython/pull/29560/files#diff-6553a99f3ae04c9fc9c2349ac27037bfe3b274d5ca54d5800d9c5b5f11e29d21R766-R774

But `self._file` could be a `BytesIO` or a `TextIOWrapper`: https://github.com/python/cpython/pull/29560/files#diff-6553a99f3ae04c9fc9c2349ac27037bfe3b274d5ca54d5800d9c5b5f11e29d21R653-R656

And according to our current stubs, `TextIOWrapper` doesn't have a `read1`, `readinto` or `readinto1` method: https://mypy-play.net/?mypy=latest&python=3.10&gist=c5b47d63657de3a0a8c9a275baf16705

Reply:
Whether or not these exist seems to depend on the mode:

```
>>> tempfile.TemporaryFile("r").read1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: '_io.TextIOWrapper' object has no attribute 'read1'
>>> tempfile.TemporaryFile("rb").read1
<built-in method read1 of _io.BufferedReader object at 0x7fb6c9dd67d0>
```

I can't think of a practical use case for `read1()` or `readinto1()`. With subprocesses, `proc.stdout.read1(100)` reads data from the stdout of a subprocess without waiting for more data to arrive (#6153). But with tempfiles, a plain `.read()` works like that and just returns empty bytes once it's at the end of the file.

Reply:
Adding a comment.